viewing_party/party.py

```diff
@@ -1,23 +1,145 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    if not (title and genre and rating):
+        return None
+    else:
+        movie = { "title" : title,
+                    "genre" : genre,
+                    "rating" : rating}
+        return movie
+    
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+
+    return user_data    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    if not user_data["watched"]:
+        return 0.0
+    else:
+        total_rating = sum(movie["rating"] for movie in user_data["watched"])
+        return total_rating / len(user_data["watched"])
+    
+def get_most_watched_genre(user_data):
+
+    if not user_data["watched"]:
+        return None
+    
+    genre_count = {}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_count:
+            genre_count[genre] += 1
+        else:
+            genre_count[genre] = 1
+    return max(genre_count, key=lambda genre: genre_count[genre])
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+
+    user_movies = user_data["watched"]
+    friend_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movies.append(movie)
+
+    unique_movies = []
+
+    for user_movie in user_movies:
+        if user_movie not in friend_movies and user_movie not in unique_movies:
+            unique_movies.append(user_movie)
+
+    print(f"{unique_movies = }")
+    return unique_movies
+
+    # user_movies = set([movie['title'] for movie in user_data['watched']])
+    # friends_movies = set([movie['title'] for friend in user_data['friends'] for movie in friend['watched']])
+    # unique_movies = user_movies.difference(friends_movies)
+    # unique_movies = [{'title': movie} for movie in unique_movies]
+    # print(f"{unique_movies = }")
+    # return unique_movies
+
+def get_friends_unique_watched(user_data):
+    user_movies = user_data["watched"]
+    friend_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movies.append(movie)

```

Comment:
Once again, you could add a quick check to see if the movie is already in "friend_movies" or not. This prevents duplicates which could avoid the compound conditional below!